### DISCUSSION TITLE
How to declare Motor parameters using block ?

### DISCUSSION THREAD

**Discussion by GenghisHan-op** [pybricks/support#1916] [General]
I'm moving my code from Python to Block, please tell me how to declare all parameters of the Motor using block ?
ClassMotor(port, positive_direction=Direction.CLOCKWISE, gears=None, reset_angle=True, profile=None)
Thank you

**Comment by BertLindeman** [comment]
Do you need all these parameters or are the defaults OK?

Obvious you need Port and in the setup block you define the positive_direction.


The doc of the reset_angle argument:
 - reset_angle (bool)
Choose True to reset the rotation sensor value to the absolute marker angle (between -180 and 179). 
  Choose False to keep the current value, so your program knows where it left off last time.

As we "prooved" in #1913 the default reset_angle is ~~'False'~~ The default actually is `True`.

Bert

**Reply by GenghisHan-op** [reply]
I need to add reset_angle parameter "False" to the motor. I can do this easily with python, but can you tell me how to do in block coding ? In the block, I can only choose port and direction for the motor.

**Reply by BertLindeman** [reply]
I do not know of an easy way to import a small python program to do this.

And I do not know if there is an expanded motor block in the planning. @laurensvalk DO you know?

**Comment by laurensvalk** [comment]
This feature isn't available in blocks. We try to strike a balance between usability and simplicity. People who use an advanced feature like this have usually already made the jump to Python.

But if you must, you _can_ do it by opening this program: [motor_with_reset.zip](https://github.com/user-attachments/files/17632617/motor_with_reset.zip)

![image](https://github.com/user-attachments/assets/7bae57f7-540c-421f-8753-1d704b73fd7a)

**Reply by GenghisHan-op** [reply]
Thanks a looooot for this.
I wrote the code with Python, after that translated to Block so that I can teach my son (3rd grade).
This means alot for me.

**Reply by BertLindeman** [reply]
Thanks Laurens, you make it look so simple. 
Would be nice to have a box full of these nice examples 😄

**Reply by laurensvalk** [reply]
This is technically "not possible". The Port and Direction parameters aren't normally available as separate blocks on the palette. But they still exist... :smile:
